src/core/ComplexFilter/index.tsx

```diff
@@ -55,12 +55,13 @@ export default function ComplexFilter<
     getInitialValue()
   );
 
-  const [pendingValue, setPendingValue] = useState<
-    Value<DefaultMenuSelectOption, true>
-  >([]);
+  const [pendingValue, setPendingValue] = useState<DefaultMenuSelectOption[]>(
+    getInitialValue() as DefaultMenuSelectOption[]
+  );
 
   useEffect(() => {
     onChange(value);
+    setPendingValue(value as DefaultMenuSelectOption[]);
```

Comment:
This is added since we removed `setPendingValue(value)` in `handleClick()` below, but still need a centralized way to keep `value` and `pendingValue` in sync!